### DISCUSSION TITLE
是否需要在JNI_OnUnload中将stub等全局变量设为NULL?

### DISCUSSION THREAD

**Discussion by zhbzhbzhbz** [bytedance/bhook#44] [Q&A] [answered]
如题，谢谢~

**Comment by caikelun** [comment, marked as answer]
不需要。在unhook后可以将stub设置为NULL，这是为了防止之后误用。你如果hook后确认不需要unhook了，那stub值也不用保留。
这和 JNI_OnUnload 完全无关。

**Reply by zhbzhbzhbz** [reply]
> 不需要。在unhook后可以将stub设置为NULL，这是为了防止之后误用。你如果hook后确认不需要unhook了，那stub值也不用保留。 这和 JNI_OnUnload 完全无关。

谢谢回复~ 那想再多问一个问题，请问按您的习惯是否会在JNI_OnUnload里释放一些自己创建的资源来节省内存呢？谷歌了一下感觉众说纷纭，有说这个函数不会被调用的，有说app被杀了自然内存就被释放了，但还有看一些源码里也有用到它
